core/src/main/java/com/squarespace/template/plugins/platform/ContentFormatters.java

```diff
@@ -320,6 +323,44 @@ public void apply(Context ctx, Arguments args, Variables variables) throws CodeE
     }
   };
 
+  public static class HslaColorFormatter extends BaseFormatter {
+
+    public HslaColorFormatter() {
+      super("hslaColor", false);
+    }
+
+    @Override
+    public void apply(Context ctx, Arguments args, Variables variables) throws CodeExecuteException {
+      Variable var = variables.first();
+      JsonNode node = var.node();
+
```

Comment:
Are you planning to clamp all of the values (hue, saturation) to their valid ranges?

Reply:
yes, we want to clamp these and produce an error message if the values fall out of bounds.   looking at `HeightFormatter`, it sets an error message the following way: `var.set("Invalid source parameter. Pass in 'originalSize'.")`.  i also see other formatters, like `VideoFormatter`, throw an `ArgumentsException` instead.  what is the recommended way to produce an error message in template-compiler?